worlds/sc2/client.py

```diff
@@ -519,6 +521,29 @@ def color_code(self, code: str) -> str:
         return '<c val="' + self.color_codes[code] + '">'
 
 
+class SC2JSONtoKivyParser(KivyJSONtoTextParser):
+    def _handle_item_name(self, node: JSONMessagePart):
+        item_name = node["text"]
+        if not item_name in item_descriptions:
+            return super()._handle_item_name(node)
+        
+        flags = node.get("flags", 0)
+        item_types = []
+        if flags & 0b001:  # advancement
```

Comment:
I'd use the name from `ItemClassification`
```python
if flags & ItemClassification.progression:
```

Reply:
Done.